### DISCUSSION TITLE
Tooltip for table columns

### DISCUSSION THREAD

**Discussion by shamilh2o** [h2oai/wave#1784] [Q&A] [answered]
Hi Team.
Is there any possibility to add column wise tooltips for a wave table instead of adding a single tooltip to a whole table?
I have a requirement on adding tooltip only for a specific column and looking how to overcome it. 
Thanks in advance.

**Comment by marek-mihok** [comment]
Hi Shamil, this is not supported in Wave currently. Is using the column-specific tooltip necessary? Could you please provide us the use case? Thank you.

**Reply by shamilh2o** [reply]
Hey @marek-mihok. Thanks for the response.

I wanna give brief description about the data what we have shown in a particular column to the user.
If it's not yet available I'll open a feature request.

**Reply by marek-mihok** [reply]
@shamilh2o the question is whether it makes sense from the UX perspective. This behaviour is not common - column headers in general are being used for sort/filter actions only. Could you please be more specific whether using the legend in the table tooltip or simply the self-explanatory column name is not enough?

However, if there is no other way, it should be [possible to achieve](https://stackoverflow.com/questions/53321904/how-to-render-a-column-header-with-both-text-and-a-tooltip-icon-in-fabric-detail):
![image](https://user-images.githubusercontent.com/23740173/211520632-4a9d166a-ca1a-451b-adde-eef70a9332fb.png)

Wdyt @mturoci?

**Reply by shamilh2o** [reply]
@marek-mihok the self-explanatory column name is not enough. The column is regarding a cost value. I need to give an idea on how the value is calculated and how often these data gonna be changed by a single sentence.  

What do you mean by "the legend in the table tooltip"? How it looks like?

**Comment by mturoci** [comment, marked as answer]
* If you want to show a tooltip for a column name (like in the picture @marek-mihok posted above), file a feature request. That's a simple thing to do.
* If you want to have a tooltip per col cell and display the whole cell content in the tooltip, you can set https://wave.h2o.ai/docs/widgets/form/table/#with-text-overflow to `tooltip`.
* If you want to have a tooltip per col cell, but tooltipText != cellText, use a master-detail view pattern instead.

**Reply by shamilh2o** [reply]
Thanks @mturoci. Will create a feature request.
